pkg/registry/apis/ofrep/register.go

```diff
@@ -324,22 +326,17 @@ func (b *APIBuilder) validateNamespace(r *http.Request) bool {
 		namespace = mux.Vars(r)["namespace"]
 	}
 
-	info, err := types.ParseNamespace(namespace)
-	if err != nil {
-		b.logger.Error("Error parsing namespace", "error", err)
-		return false
-	}
-
-	// Extract stackId from feature flag evaluation context
+	// Extract namespace from feature flag evaluation context
 	body, err := io.ReadAll(r.Body)
 	if err != nil {
 		b.logger.Error("Error reading evaluation request body", "error", err)
 		return false
 	}
 	r.Body = io.NopCloser(bytes.NewBuffer(body))
 
+	evalCtxNamespace := b.namespaceFromEvalCtx(body)
 	// "default" namespace case can only occur in on-prem grafana
-	if b.stackIdFromEvalCtx(body) == info.StackID {
+	if (namespace == "default" && len(evalCtxNamespace) == 0) || (evalCtxNamespace == namespace) {
```

Comment:
`s == ""` is much clearer than `len(s) == 0` 😉

```suggestion
	if (namespace == "default" && evalCtxNamespace == "") || (evalCtxNamespace == namespace) {
```